src/LinuxTracing/UprobesUnwindingVisitorTest.cpp

```diff
@@ -1639,4 +1511,205 @@ TEST_F(
   EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
 }
 
+class UprobesUnwindingVisitorMmapTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    ON_CALL(maps_, Find).WillByDefault([this](uint64_t pc) { return real_maps_->Find(pc); });
+    ON_CALL(maps_, Get).WillByDefault([this]() { return real_maps_->Get(); });
+    ON_CALL(maps_, AddAndSort)
+        .WillByDefault([this](uint64_t start, uint64_t end, uint64_t offset, uint64_t flags,
+                              const std::string& name) {
+          return real_maps_->AddAndSort(start, end, offset, flags, name);
+        });
+  }
+
+  static constexpr pid_t kPid = 42;
+
+  MockTracerListener listener_;
+  MockLibunwindstackMaps maps_;
+  UprobesUnwindingVisitor visitor_{&listener_,
+                                   &function_call_manager_,
+                                   &return_address_manager_,
+                                   &maps_,
+                                   &unwinder_,
+                                   &leaf_function_call_manager_,
+                                   /*user_space_instrumentation_addresses=*/nullptr};
+
+ private:
+  UprobesFunctionCallManager function_call_manager_;
+  MockUprobesReturnAddressManager return_address_manager_{
+      /*user_space_instrumentation_addresses=*/nullptr};
+  std::unique_ptr<LibunwindstackMaps> real_maps_ = LibunwindstackMaps::ParseMaps("");
+  MockLibunwindstackUnwinder unwinder_;
+  MockLeafFunctionCallManager leaf_function_call_manager_{128};
+};
+
+TEST_F(UprobesUnwindingVisitorMmapTest,
+       VisitMmapPerfEventUpdatesLibunwindstackMapsAndSendsModuleUpdates) {
```

Comment:
This test is mostly unchanged. I moved it down and it now uses the new fixture. The new test is the one below.